controllers/cm_event_handler.go

```diff
@@ -0,0 +1,70 @@
+package controllers
+
+import (
+	"context"
+
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+)
+
+type ConfigMapEventHandler struct {
+	reconciler *KataConfigOpenShiftReconciler
+}
+
+func (ch *ConfigMapEventHandler) Create(ctx context.Context, event event.CreateEvent, queue workqueue.RateLimitingInterface) {
+
+	if ch.reconciler.kataConfig == nil {
+		return
+	}
+
+	cm := event.Object
+
+	// Check if the configMap name is relevant to the operator
+	if cm.GetNamespace() != OperatorNamespace || !isConfigMapRelevant(cm.GetName()) {
+		return
+	}
+	log := ch.reconciler.Log.WithName("CMCreate").WithValues("cm name", cm.GetName())
+	log.Info("FeatureGates configMap created")
+
+	queue.Add(ch.reconciler.makeReconcileRequest())
+}
+
+func (ch *ConfigMapEventHandler) Update(ctx context.Context, event event.UpdateEvent, queue workqueue.RateLimitingInterface) {
+
+	if ch.reconciler.kataConfig == nil {
+		return
+	}
+
+	cm := event.ObjectNew
+
+	// Check if the configMap name is relevant to the operator
+	if cm.GetNamespace() != OperatorNamespace || !isConfigMapRelevant(cm.GetName()) {
+		return
+	}
```

Comment:
We might want to check here whether the `spec` subobjects of the old and new configmap versions actually differ, otherwise we end up reconciliating on spurious metadata or status changes.

Reply:
I see..